backend/api/groups.py

```diff
@@ -86,3 +86,50 @@ async def get_groups(
                      for group in groups
                      if user.id in group["member_ids"]]
     return group_objects
+
+
+@router.put("/{group_id}", status_code=status.HTTP_200_OK,
+            response_model=models.Group)
+async def update_group(
+        group_id: PydanticObjectId,
+        request: models.Group,
+        user: Annotated[models.User, Depends(auth.get_current_user)]
+):
+    """
+    # Update group by id
+    Updates a group by its id in the database.
+    ```
+    Args:
+        group_id (str): The id of the group to update.
+        request (models.Group): The updated group information.
+        user (models.User): The authenticated user.
+    ```
+    """
+    group_dict = db["groups"].find_one({"_id": group_id})
+    if not group_dict:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND,
+                            detail="Group not found")
+    group = models.Group(**group_dict)
+    if user.id != group.owner_id:
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN,
+                            detail="User is not the owner of the group")
```

Comment:
I think we should take a more user-oriented approach: e.g., "You are not the owner of the group" - to be discussed